http-clients/netty-nio-client/src/main/java/software/amazon/awssdk/http/nio/netty/internal/http2/MultiplexedChannelRecord.java

```diff
@@ -184,7 +185,19 @@ void closeChildChannels() {
      * Delivers the exception to all registered child channels, and prohibits new streams being created on this connection.
      */
     void closeChildChannels(Throwable t) {
-        closeAndExecuteOnChildChannels(ch -> ch.pipeline().fireExceptionCaught(t));
+        closeAndExecuteOnChildChannels(ch -> {
+            RuntimeException ioException = new RuntimeException("An exception occurred on the connection ", t);
+
+            // Deliver the exception only if the channel knows to handle the exception
+            if (ch.pipeline().get(ResponseHandler.class) != null) {
+                log.debug(() -> "Delivering exception to child channel " + ch + ioException);
+                ch.pipeline().fireExceptionCaught(ioException);
+            } else {
+                log.debug(() -> "Closing child channel " + ch + ioException);
+                ch.close();
```

Comment:
This seems super fragile. Can we just have a handler at the end of every pipeline that closes on exception?

Reply:
The connection could get closed before any handler is attached to stream channels and there would be nothing catching the exception. I found out this issue in my testing where I set a very low `pingPeriod`.

Reply:
Can we have a handler attached to the end of every channel as soon as it is created, before it is even connected, that calls `close()` when an exception is fired on the channel, assuming nothing else closed the channel already?

Reply:
Updated to attach `UnusedChannelExceptionHandler` to the stream channels once they are created and that should take care of it.